src/scipp/core/assignments.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from __future__ import annotations
+
+from typing import Dict, Union
+
+from ..typing import DataArray, Dataset
+
+
+def assign_coords(self, coords: Dict) -> Union[DataArray, Dataset]:
```

Comment:
Why did you remove the keyword-arg syntax?

Reply:
> `coords` should be position-only, such that it does not prevent the use of a keyword arg of that name.

Sorry I misunderstood this comment. Then... what do you mean by `position-only` here ...?

Reply:
I meant the dict should be position-only, but keep the keyword args.

Reply:
```python
def assign_coords(self, /, coords: Dict[str, Variable], **kwargs):
```
or something like that

Reply:
Ah, I see. My question was more like, should we allow only either of them

More specifically, will this be allowed?
```
da.assign_coords({'coord0': coord0_a}, coord0=coord0_b)
```
Currently, the keyword argument will overwrite the positional argument in this case.